osgar/node.py

```diff
@@ -0,0 +1,46 @@
+"""
+  Node - processing unit
+"""
+from threading import Thread
+
+from osgar.bus import BusShutdownException
+
+
+class Node(Thread):
+    """
+       Parent classs for all processing "nodes".
+    """
+    def __init__(self, config, bus):
+        super().__init__()
+        self.setDaemon(True)
+        self.bus = bus
+        self.time = None
+
+    def publish(self, channel, data):
+        return self.bus.publish(channel, data)
+
+    def listen(self):
+        return self.bus.listen()
+
+    def sleep(self, secs):
+        self.bus.sleep(secs)
+
+    def is_alive(self):
+        return self.bus._is_alive
+
+    def update(self):
+        timestamp, channel, data = self.bus.listen()
+        self.time = timestamp
+        setattr(Node, channel, data)
+
+    def run(self):
+        try:
+            while True:
+                self.update()
```

Comment:
I think only this inner part of the `run` method should be overidable by the user. The outer part (try/except) should stay part of the infrastructure.

Reply:
I agree, but I am not sure how to enforce it? Or do you want to use different name for `run()`?

Reply:
Maybe one more note regarding try/except in `MyTimer` ... it is not needed there unless `publish` can rise Exception. And it can also be implemented via `update()`.